File: src/net/merayen/elastic/ui/Point.java
package net.merayen.elastic.ui;

public class Point {
	public float x;
	public float y;

	public Point() {
		this.x = 0f;
		this.y = 0f;
	}

	public Point(Point p) {
		this.x = p.x;
		this.y = p.y;
	}

	public Point(float x, float y) {
		this.x = x;
		this.y = y;
	}
	
	public String toString() {
		return String.format(
				"[Point(x=%f, y=%f)]",
				x, y
		);
	}
}

File: src/main/kotlin/net/merayen/elastic/util/Point.kt
package net.merayen.elastic.util

class Point {
	var x = 0f
	var y = 0f

	constructor() {
		x = 0f
		y = 0f
	}

	constructor(p: Point) {
		x = p.x
		y = p.y
	}

	constructor(x: Float, y: Float) {
		this.x = x
		this.y = y
	}

	override fun toString() = "[Point(x=$x, y=$y)]"
}
